lib/array_equals.rb

```diff
@@ -1,5 +1,26 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+
+  # check if any arrays == nil
+  check_nils_arr = [array1, array2]
+  no_nil = true
+  check_nils_arr.length.times do |index|
+    no_nil = false if check_nils_arr[index] == nil
+  end
+
+  is_same = true
+  if no_nil
+    if array2.length != array1.length
+      is_same = false
```

Comment:
You can just `return False` here instead of saving into a variable. That way your code execution ends here since you already know that the arrays are not equal to each other.